src/validation.cpp

```diff
@@ -996,7 +996,7 @@ bool MemPoolAccept::PreChecks(ATMPArgs& args, Workspace& ws)
     // already calculated.
     if (const auto err{SingleV3Checks(ws.m_ptx, ws.m_ancestors, ws.m_conflicts, ws.m_vsize)}) {
         // Disabled within package validation.
-        if (err->second != nullptr && args.m_allow_replacement) {
+        if (err->second != nullptr && args.m_allow_replacement && !args.m_package_feerates) {
```

Comment:
`m_package_feerates` is experiencing mission creep here. Really this is asking "was I called in AcceptSingleTransaction"?

Should we bikeshed this to make it clearer @glozow ?

Reply:
Agree there's room for improvement clarity-wise.

Perhaps a good approach is to have each flag control 1 part of the validation logic (even if 1 flag implies another flag)
- whether you call `LimitMempoolSize()` within `Finalize()`
- whether all conflicts should be rejected immediately
- whether `CheckFeeRate` should be skipped in `PreChecks`
- whether sibling eviction can be considered
- whether package RBF can be considered

The `ATMPArgs` constructors should decide the flags based on where they are created and we can add a sanity checking function within each ctor to assert that package RBF implies package feerates, package feerates implies package submission, etc.

Reply:
added an explicit `m_allow_sibling_eviction` ATMPArg and had it being true imply `m_allow_replacements`.